JotunnLib/Managers/ZoneManager.cs

```diff
@@ -1,47 +1,300 @@
+        /// <param name="fixLocationReferences">Replace JVLmock GameObjects with a copy of their real prefab</param>
+        /// <returns></returns>
+        public GameObject CreateLocationContainer(GameObject gameObject, bool fixLocationReferences = false)
+        {
+            var locationContainer = Object.Instantiate(gameObject, LocationContainer.transform);
+            if (fixLocationReferences)
+            {
+                var transform = locationContainer.transform;
+                FixMockReferences(transform);
+            }
             
-            Logger.LogInfo("Registering custom zone data");
+            return locationContainer;
+        }
 
-            // Call event handlers to load prefabs
-            OnBeforeZoneLoad?.Invoke(null, EventArgs.Empty);
+        private void FixMockReferences(Transform transform)
+        {
+            for (int i = 0; i < transform.childCount; i++)
+            {
+                var child = transform.GetChild(i);
+                if (!child.name.StartsWith(MockManager.JVLMockPrefix))
+                {
+                    //Allow nested component references to JVLmock
+                    FixMockReferences(child);
+                    continue;
+                }
+                string prefabName = child.name.Substring(MockManager.JVLMockPrefix.Length);
+
+                //Allow duplicated JVLmocks (child names must be unique)
+                Match match = copyRegex.Match(prefabName);
+                if (match.Success)
+                {
+                    prefabName = prefabName.Substring(0, match.Index);
+                }
+
+                var replacementPrefab = PrefabManager.Instance.GetPrefab(prefabName);
+                if (!replacementPrefab)
+                {
+                    Logger.LogWarning($"No replacement prefab found for {prefabName}");
+                }
+                else
+                {
+                    var replacement = Object.Instantiate(replacementPrefab, child.localPosition, child.localRotation, transform);
+                    replacement.transform.localScale = child.localScale;
+                    Object.Destroy(child.gameObject);
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Create a CustomLocation that is a deep copy of the original.
+        ///     Changes will not affect the original. The CustomLocation is already registered to be added.
+        /// </summary>
+        /// <param name="name">name of the custom location</param>
+        /// <param name="baseName">name of the existing location to copy</param>
+        /// <returns></returns>
+        public CustomLocation CreateClonedLocation(string name, string baseName)
+        {
+            var baseZoneLocation = GetZoneLocation(baseName);
+            var copiedPrefab = Object.Instantiate(baseZoneLocation.m_prefab, Vector3.zero, Quaternion.identity, LocationContainer.transform);
```

Comment:
Should use PrefabManager, i guess

Reply:
Locations aren't retrieved from ZNetScene, but are in the ZoneSystem. Vegetation needs to be in the PrefabManager though. Probably not an issue to add it though, pretty much the same. Another opportunity for name collisions though.